### PR TITLE
Scale only labeled pods

### PR DESCRIPTION
**Pull request by ca16**
I added logic that allows you to tell the autoscaler to only scale pods that have the label `scale-for-me=yes` by passing the flag `scale-labeled-only`. 

The default (the flag is not passed) means it will try to scale for anything. 

This is a very basic version of filtering using a flag - I'm definitely open to suggestions regarding changing what the label should be called, or even using options to specify a label to scale by instead of having one constant one.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -45,11 +45,12 @@
                    "for more verbosity.",
               type=click.IntRange(0, 3, clamp=True),
               count=True)
+@click.option("--scale-labeled-only", is_flag=True)
```
Comment (jkinkead) [inline main.py]: I would make this a string-valued flag (`--scale-label=autoscale-job`). Then the presence of the label on the task (regardless of value) would trigger a rescale.

**Review by jkinkead — COMMENTED** [review]
One suggestion.

File: main.py (hunk 1 above)
Comment (ca16) [reply main.py]: Sounds good.

Comment (ca16) [reply main.py]: To clarify: the value someone passes to the flag is the label the autoscaler should see on a pod in order to consider scaling up for that pod?

Comment (jkinkead) [reply main.py]: Yes, that's what I was suggesting. :)

**Comment by ca16** [comment]
Changes made!

File: README.md [hunk 2]
```diff
@@ -133,6 +133,7 @@ primary region and "us-east-1" as the secondary.
 - --slack-hook: Optional [Slack incoming webhook](https://api.slack.com/incoming-webhooks) for scaling notifications
 - --dry-run: Flag for testing so resources aren't actually modified. Actions will instead be logged only.
 - -v: Sets the verbosity. Specify multiple times for more log output, e.g. `-vvv`
+- --scale-label: If you want the autoscaler to scale up for only some pods, add a label to those pods and pass it to `--scale-label`. If this is not included, the autoscaler will consider scaling up for any pod. E.g. including `--scale-label autoscaler-task` means that the autoscaler will consider scaling up only for pods labeled with `autoscaler-task`.
```
Comment (jkinkead) [inline README.md]: Rephrase:
```
Optional label to look for on pods. If this flag is set, the autoscaler will only count
unscheduled pods with the given label when determining if more nodes are needed.
```

**Review by jkinkead — COMMENTED** [review]
Suggestion to shorten & simplify flag documentation.

File: README.md (hunk 2 above)
Comment (ca16) [reply README.md]: Yeah that sounds better.

**Comment by ca16** [comment]
Updated documentation

**Review by jkinkead — APPROVED** [review]
LGTM!

**Pull request merged**